src/haz3lcore/statics/Self.re

```diff
@@ -35,6 +35,7 @@ type t =
 [@deriving (show({with_path: false}), sexp, yojson)]
 type exp =
   | Free(Var.t)
+  | InexhaustiveMatch(exp)
```

Comment:
`Self.InexhaustiveMatch` is finally designed to be a "wrapper" around (`Self.exp` assigned to a match expression when exhaustiveness checking is ignored). I wanted it to be as flexible as possible so that it might be easier to add new features to the match expression in the future.